File: JavaDemo/src/work-spring/acmicpc/src/main/java/com/grepp/boot/model/repository/UserRepository.java
package com.grepp.boot.model.repository;

import com.grepp.boot.model.dto.UserDTO;
import com.grepp.boot.model.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

//@Mapper
@Repository
public interface UserRepository extends JpaRepository<UserEntity, String> {

//    @Select("SELECT user_id, user_pw FROM users")
//    List<UserDTO> findAll();

//    @Select("SELECT user_id, user_pw FROM users WHERE user_id = #{userId}")
//    UserDTO findById(String userId);

    // 유저 등록
//    @Insert("INSERT INTO users (user_id, user_pw) VALUES (#{userId}, #{userPw})")
//    int register(UserDTO userDTO);

    // 유저 가져오기 (userId)
    Optional<UserEntity> findByUserId(String userId);

}

File: JavaDemo/src/work-kotlin/day1021_todo/src/main/kotlin/com/grepp/todo/model/repository/UserRepository.kt
package com.grepp.todo.model.repository

import com.grepp.todo.model.entity.UserEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository: CrudRepository<UserEntity, Long> {
    // 회원가입 함수는 자동으로 create 만들어지니까 로그인 함수만 만들어 놓음
    fun findByUsernameAndPassword(username: String, password: String): UserEntity?  // 로그인 실패 시 결과 null 허용되게 ? 붙임
}